app/Http/Controllers/Internlink/InternlinkController.php

```diff
@@ -160,80 +160,83 @@ public function viewAll()
         return view('internlink.results')->with('internships', $internships);
     }
 
-    // $option can be number or 'contact'
-    public function viewProfile($option)
+    public function myContact()
     {
-        $member_id = Auth::user()->member->id;
-        $link = Link::where('member_id', $member_id)->get()->first();
-        if($link === null)
-        {
+        $form = $this->form(InternlinkSignupForm::class, [
+            'method' => 'POST',
+            'url' => route('internlink.profile.contact.update'),
+            'model' => Auth::user()->member->internlink,
+        ]);
+
+        return $this->profile(Link::rules(), $form);
+    }
+
+    public function myInternship($id)
+    {
+        if (Auth::user()->member->internlink === null) {
```

Comment:
I didn't want an error page to appear if someone manually edited the URL. But then again, that is outside the normal range of expected behaviors so I think its OK.